cmd/dbos/cli_integration_test.go

```diff
@@ -222,32 +261,59 @@ func testProjectInitialization(t *testing.T, cliPath string) {
 	require.NoError(t, err, "go mod tidy failed: %s", string(modOutput))
 }
 
+func testMigrateCommand(t *testing.T, cliPath string, baseArgs []string, dbRole string) {
+	// If the role is set, create it in Postgres first
+	if dbRole != "postgres" {
+		db, err := sql.Open("pgx", getDatabaseURL("postgres"))
+		require.NoError(t, err)
+		defer db.Close()
+		password := os.Getenv("PGPASSWORD")
+		if password == "" {
+			password = "dbos"
+		}
+		query := fmt.Sprintf("CREATE ROLE %s LOGIN PASSWORD '%s'", pgx.Identifier{dbRole}.Sanitize(), password)
+		_, err = db.Exec(query)
+		require.NoError(t, err)
+	}
+
+	args := append([]string{"--verbose", "migrate"}, baseArgs...)
+	if dbRole != "postgres" {
+		args = append(args, "--app-role", dbRole)
+	}
+	cmd := exec.Command(cliPath, args...)
+	cmd.Env = append(os.Environ(), "DBOS_SYSTEM_DATABASE_URL="+getDatabaseURL("postgres"))
+	output, err := cmd.CombinedOutput()
+	require.NoError(t, err, "Migrate command failed: %s", string(output))
+	assert.Contains(t, string(output), "DBOS migrations completed successfully", "Output should confirm migration")
+	fmt.Println(string(output))
```

Comment:
We also need to test the app actually works with the new role after the migration

Reply:
The other tests do that (they'll use the role that's passed to the suite)